Assets/PlayroomKit/src/index.js

```diff
@@ -122,13 +125,52 @@ mergeInto(LibraryManager.library, {
       return;
     }
 
-    Playroom.onPlayerJoin((player) => {
-      var id = player.id;
-      var bufferSize = lengthBytesUTF8(id) + 1;
-      var buffer = _malloc(bufferSize);
-      stringToUTF8(id, buffer, bufferSize);
-      dynCall("vi", functionPtr, [buffer]);
-    });
+    var callbackID = Date.now().toString();
+    try {
+      var unsubcribePlayerJoin = Playroom.onPlayerJoin((player) => {
+        var id = player.id;
+        var bufferSize = lengthBytesUTF8(id) + 1;
+        var buffer = _malloc(bufferSize);
+        stringToUTF8(id, buffer, bufferSize);
+
+        dynCall("vi", functionPtr, [buffer]);
+      });
+    } catch (error) {
+      console.log(error);
+    }
+
+
+    this.onPlayerJoinCallBacks[callbackID] = unsubcribePlayerJoin;
+    var callbackIDbufferSize = lengthBytesUTF8(callbackID) + 1;
+    var callbackIDUTF8 = _malloc(callbackIDbufferSize);
+    stringToUTF8(callbackID, callbackIDUTF8, callbackIDbufferSize);
+    return callbackIDUTF8;
+
+  },
+
+  UnsubscribeOnPlayerJoinInternal: function (id) {
+    functionId = UTF8ToString(id);
+    var unsubscribeFunction = this.onPlayerJoinCallBacks[functionId];
+    if (unsubscribeFunction) {
+      unsubscribeFunction();
+      delete this.onPlayerJoinCallBacks[functionId];
+    } else {
+      console.error("No player join event handler with ID " + functionId + " to unregister.");
+    }
+  },
+
+  UnsubscribeOnQuitInternal: function () {
```

Comment:
Are we planning to do the same with `UnsubscribeOnQuitInternal` as we've done with `UnsubscribeOnPlayerJoinInternal`?

Reply:
Yes

Reply:
Test the current implementation for onPlayerJoin. If all OK, replicate for onQuit quick and let's close this down. 

Otherwise, you can choose to merge this PR just for onPlayerJoin, then create a separate PR for onQuit. 

Your call!